### ISSUE TITLE
3.3.8: Ensuring single transactions (from the user's perspective) are covered

### ISSUE THREAD

**Issue by benja11y** [w3c/wcag#1880]
There are several sites on which, in order to complete a transaction such as buying something, the user is redirected to a payment provider before being sent back to the original site. E.g. when buying something from eBay, some of the screens in the middle of this process will be provided by PayPal.

We understand that such screens _would_ be covered here, as from the user's perspective they form part of one single transaction.

This was the case with the initial version of the wording, and we gather it would be the case with the latest wording, which includes the term "user-session". The term is not defined, and could have many different meanings based on who is interpreting the SC, so we suggest it be removed as it appears to be redundant.

This issue has been raised on behalf of [TPGi](https://www.tpgi.com), based on an internal consultation process. [An article containing all of our comments](https://www.tpgi.com/wcag-22-working-draft-feedback/) can be found on our blog, along with an introduction and acknowledgments.

**Comment by alastc** [comment]
> We understand that such screens would be covered here, as from the user's perspective they form part of one single transaction.

In WCAG terms, I think it depends that they are part of the same process (yes), and same user-session (technically no, but possibly yes from a user point of view?). 

I'm not sure whether you are saying that it should be covered or not? 

Assuming we further define "user-session", should that include technical aspects such as whether it is on the same domain, or take it from a user point of view? (Question for the group.)

**Comment by DavidMacDonald** [comment]
Under the current WCAG 2.x conformance schema, conformance is determined by a (URL) page by page basis, and so a part of the process that is sent to another URL owned by a third party could get complicated. 

I doubt a company has the authority to claim a conformance to a url that they do not own.

If however, those parts of the process are injected into the page URL with AJAX or an iframe (etc.) it is still at the same URL and would be in scope.
@alastc @benja11y

**Comment by awkawk** [comment]
I agree with David's and Alastair's analysis - in order for a page at example.com where you add something to the cart and click "purchase" to pass, all pages that are part of the checkout process must pass at a certain level to make a conformance claim at that level. I do not think that it matters if all of the steps are in the example.com domain, if some are at paypal.com, or if there is some iframe injection of paypal.com content into the example.com page.

This raises an issue that we may not have considered, which is that the different systems are less likely to exchange information about the user, and there may be privacy rules that would prevent the sharing of information about the shopper from the store site with the transaction site. These systems are separate and while it may be possible for the information to be shared we should make sure that we get the opinions of people who deal with this topic to provide input on possible problems.

**Comment by alastc** [comment]
I think this is a case were there is a circumstance where process & session do not align, at least from a technical reading of 'session'.

1. A site with a contained checkout is one process and one session.
2. A site with a 3rd party payment/checkout in an iframe is one process and one session.
3. A site with a separate 3rd party payment on a separate domain is one process but **two sessions**.

That is a technical reading of 'session', but given that web-developers are a large constituent of the readership it will be a common one: "a series of contiguous actions by a visitor on an **individual website** within a given time frame." from https://hazelcast.com/glossary/web-session/

Therefore in that 3rd scenario it would be out of scope due to the separate session.

I think there are two paths:

1. Remove session and keep the scope to process (probably better aligns with our conformance model).
2. Keep session note the 3rd scenario in the understanding doc, saying that it is out of scope for this SC.

The best path might depend on feedback from payment providers, i.e. is it feasible for their customers to provide information to the provider so that it is not asked for twice?

**Comment by alastc** [comment]
So the result of the conversation was that:

- The group preferred the route of focusing on process so that 3rd party providers would be in scope.
- The term session was removed from the SC text, BUT, the explanation about that was included in the understanding doc.
- There is an addition about 3rd parties:

This Success Criterion does not add a requirement to remember information between sessions. A <a>process</a> is defined on the basis of an activity and is not applicable when a user returns after closing a session or navigating away. **However, a process can run across different domains, so if a check-out process includes a 3rd party payment provider, that would be in scope.**
